### PR TITLE
Automation test for Components

### PR DESCRIPTION
**Pull request by NipunaMarcus**

### CODE REVIEW

File: modules/web/webpack.config.js [hunk 1]
```diff
@@ -77,10 +78,23 @@ const config = [{
             $: 'jquery',
             jQuery: 'jquery',
         }),
+        /*
+        new CircularDependencyPlugin({
+            exclude: /a\.css|node_modules/,
+            failOnError: true,
+        }),
+        */
```
Comment (nadeeshaan) [inline modules/web/webpack.config.js]: Any specific reason for keeping this particular commented code?

Comment (NipunaMarcus) [inline modules/web/webpack.config.js]: for the future to find the circular dependencies ... lets keep it for the moment.

File: modules/web/react-test/__tests__/source.gen.test.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    // beforeEach((beforeAllDone) => {
+    //
+    //         .then(() => beforeAllDone())
+    //         .catch(beforeAllDone);
+    // });
+    fetchConfigs();
+    const testResDir = path.resolve(path.join(directory, 'js', 'tests', 'resources'));
+    loadFiles(testResDir);
+
+    it('renders correctly', () => {
```
Comment (roonyh) [inline modules/web/react-test/__tests__/source.gen.test.js]: This should an asynchronous test. Either use the done callback or return the promise. https://mochajs.org/#asynchronous-code.

File: modules/web/react-test/__tests__/source.gen.test.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    return fs.readFileSync(filePath, 'utf8');
+}
+
+describe('Source Generation Test Suite', function () {
+    // beforeEach((beforeAllDone) => {
+    //
+    //         .then(() => beforeAllDone())
+    //         .catch(beforeAllDone);
+    // });
+    fetchConfigs();
```
Comment (roonyh) [inline modules/web/react-test/__tests__/source.gen.test.js]: This is an async function right? Should be put in a before block with 'done' callback.

Comment (NipunaMarcus) [inline modules/web/react-test/__tests__/source.gen.test.js]: yeah sure ... this test is not yet good to go. I'll remove it

**Pull request merged**